eyelid-match-ops/src/primitives/yashe.rs

```diff
@@ -104,14 +130,93 @@ where
         (priv_key, pub_key)
     }
 
+    /// Encrypt a message m encoded in the polynomial ring
+    pub fn encrypt(
+        &self,
+        mut m: Message<C>,
+        public_key: PublicKey<C>,
+        rng: &mut ThreadRng,
+    ) -> Ciphertext<C> {
+        // TODO: document the equations that are being implemented by each block.
+
+        // Create the ciphertext by sampling error polynomials and applying them to the public key.
+        let s = self.sample_err(rng);
+        let e = self.sample_err(rng);
+        let mut c = s * public_key.h + e;
+
+        // Divide the polynomial coefficient modulus by T, using primitive integer arithmetic.
+        let qdt = C::modulus_as_u128() / C::t_as_u128();
+        let qdt = C::Coeff::from(qdt);
+
+        // Multiply the message by the qdt scalar, and add it to the ciphertext.
+        m.m *= qdt;
+        c += m.m;
```

Comment:
Please check that these are the equations we intend to implement for encryption.

Reply:
This is correct.